protocol/receiveversion/receiveversion.go

```diff
@@ -0,0 +1,79 @@
+package receiveversion
+
+import (
+	"github.com/kaspanet/kaspad/blockdag"
+	"github.com/kaspanet/kaspad/netadapter/router"
+	peerpkg "github.com/kaspanet/kaspad/protocol/peer"
+	"github.com/kaspanet/kaspad/wire"
+	"github.com/pkg/errors"
+)
+
+var (
+	// allowSelfConnections is only used to allow the tests to bypass the self
+	// connection detecting and disconnect logic since they intentionally
+	// do so for testing purposes.
+	allowSelfConnections bool
+
+	// minAcceptableProtocolVersion is the lowest protocol version that a
+	// connected peer may support.
+	minAcceptableProtocolVersion = wire.ProtocolVersion
+)
+
+// ReceiveVersion waits for the peer to send a version message, sends a
+// verack in response, and updates its info accordingly.
+func ReceiveVersion(incomingRoute *router.Route, outgoingRoute *router.Route, peer *peerpkg.Peer,
+	dag *blockdag.BlockDAG) (addr *wire.NetAddress, routeClosed bool, err error) {
+
+	msg, isClosed := incomingRoute.Dequeue()
```

Comment:
Please rename msg to message.

Reply:
Dequeue returns isOpen, actually.
Please go over the code and make sure you didn't make the same error anywhere else.